tests/integration/helm/upgrade/util.go

```diff
@@ -168,6 +168,64 @@ func performInPlaceUpgradeFunc(previousVersion string, isAmbient bool) func(fram
 	}
 }
 
+// upgradeAllButZtunnel returns the provided function necessary to run inside an integration test
+// for upgrading everythying but the ztunnel
+func upgradeAllButZtunnel(previousVersion string) func(framework.TestContext) {
+	return func(t framework.TestContext) {
+		isAmbient := true
+		cs := t.Clusters().Default().(*kubecluster.Cluster)
+		h := helm.New(cs.Filename())
+		nsConfig := helmtest.DefaultNamespaceConfig
+		t.CleanupConditionally(func() {
+			// only need to do call this once as helm doesn't need to remove
+			// all versions
+			helmtest.DeleteIstio(t, h, cs, nsConfig, isAmbient)
+		})
+		s := t.Settings()
+		prevVariant := s.Image.Variant
+		overrideValuesFile := helmtest.GetValuesOverrides(t, gcrHub, "", prevVariant, "", isAmbient)
+		// todo tag version is not helm version
+		helmtest.InstallIstio(t, cs, h, overrideValuesFile, previousVersion, true, isAmbient, nsConfig)
+		helmtest.VerifyInstallation(t, cs, nsConfig, true, isAmbient, "")
+
+		_, oldClient, oldServer := sanitycheck.SetupTrafficTestAmbient(t, "")
+		sanitycheck.RunTrafficTestClientServer(t, oldClient, oldServer)
```

Comment:
```suggestion
		correctnessCheck.RunTrafficTestClientServer(t, oldClient, oldServer)
```

meganit: Can we avoid the use of phrases like `sanity check` in the codebase/examples - `check/verify/correctness-check` etc are probably better.

(I'm rather bad at remembering to do this myself, which is the only reason I'm mentioning it here).